cot/src/handler.rs

```diff
@@ -59,39 +59,41 @@ pub(crate) fn into_box_request_handler<T, H: RequestHandler<T> + Send + Sync>(
 
 macro_rules! impl_request_handler {
     ($($ty:ident),*) => {
-        impl<T, $($ty,)* R> RequestHandler<($($ty,)*)> for T
+        impl<Func, $($ty,)* Fut, R> RequestHandler<($($ty,)*)> for Func
         where
-            T: FnOnce($($ty,)*) -> R + Clone + Send + Sync + 'static,
+            Func: FnOnce($($ty,)*) -> Fut + Clone + Send + Sync + 'static,
             $($ty: FromRequestParts + Send,)*
-            R: for<'a> Future<Output = Result<Response>> + Send,
+            Fut: Future<Output = R> + Send,
+            R: IntoResponse,
         {
             #[allow(non_snake_case)]
-            async fn handle(&self, request: Request) -> Result<Response> {
+            async fn handle(&self, request: Request) -> cot::Result<Response> {
```

Comment:
```suggestion
            async fn handle(&self, request: Request) -> Result<Response> {
```

It doesn't seem like there is any reason to add a path prefix (and in places there is, we prefer `crate::` to `cot::`).